hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/fees/SimpleFeeCalculatorImpl.java

```diff
@@ -110,14 +121,23 @@ public static long countKeys(@NonNull final Key key) {
      * Default implementation for query fee calculation.
      *
      * @param query The query to calculate fees for
-     * @param feeContext fee context
+     * @param queryContext the query context
      * @return Never returns normally
      * @throws UnsupportedOperationException always
      */
     @Override
     @NonNull
-    public FeeResult calculateQueryFee(@NonNull final Query query, @Nullable final FeeContext feeContext) {
-        throw new UnsupportedOperationException(
-                "Query fee calculation not supported for " + getClass().getSimpleName());
+    public FeeResult calculateQueryFee(@NonNull final Query query, @NonNull final QueryContext queryContext) {
+        final var result = new FeeResult();
+        // Add node base + extras
+        result.addNodeFee(1, feeSchedule.node().baseFee());
```

Comment:
From here, we don't need node, network and service components like transactions. All we do here is for `nodePayment`. So this should just be doing `baseFee + extras`

```
{
  "name": "FileService",
  "queries": [
    {
      "name": "FileGetContents",
      "baseFee": 0,
      "extras": [
        {
          "name": "Bytes"
        }
      ]
    },
    {
      "name": "FileGetInfo",
      "free": true
    }
  ]
}
```